File: epi/epi_judge_cpp/snake_string.cc
#include <string>

#include "test_framework/generic_test.h"
using std::string;

string SnakeString(const string& s) {
  // TODO - you fill in here.
  string result;
  for (int i = 1; i < size(s); i += 4) {
      result += s[i];
  }
  for (int i = 0; i < size(s); i += 2) {
      result += s[i];
  }
  for (int i = 3; i < size(s); i += 4) {
      result += s[i];
  }
  return result;
}

int main(int argc, char* argv[]) {
  std::vector<std::string> args{argv + 1, argv + argc};
  std::vector<std::string> param_names{"s"};
  return GenericTestMain(args, "snake_string.cc", "snake_string.tsv",
                         &SnakeString, DefaultComparator{}, param_names);
}

File: epi/epi_judge_python/snake_string.py
from test_framework import generic_test


def snake_string(s: str) -> str:
    # TODO - you fill in here.
    result = ""
    for i in range(1, len(s), 4):
        result += s[i]
    for i in range(0, len(s), 2):
        result += s[i]
    for i in range(3, len(s), 4):
        result += s[i]
    return result


if __name__ == '__main__':
    exit(
        generic_test.generic_test_main('snake_string.py', 'snake_string.tsv',
                                       snake_string))
